packages/story-editor/src/components/canvas/layout.js

```diff
@@ -380,26 +379,78 @@ const PageArea = forwardRef(function PageArea(
         zoomSetting,
         scrollLeft,
         scrollTop,
+        displayLayer,
       },
+      actions: { setScrollOffset },
     }) => ({
       hasVerticalOverflow,
       hasHorizontalOverflow,
       zoomSetting,
       scrollLeft,
       scrollTop,
+      setScrollOffset,
+      displayLayer,
     })
   );
 
+  const [scrollableLayer, setRawScrollableLayer] = useState();
+
+  const lastScrollableLayer = useRef();
+  const lastOnScroll = useRef();
+  const setScrollableLayer = useCallback(
+    (node) => {
+      setRawScrollableLayer(node);
+      // Detach last listener if we had one
+      if (lastScrollableLayer.current && lastOnScroll.current) {
+        lastScrollableLayer.current.removeEventListener(
+          'scroll',
+          lastOnScroll.current
+        );
+        lastScrollableLayer.current = null;
+        lastOnScroll.current = null;
+      }
+      if (node) {
+        // Manually attach listener so we get the event early
+        const onScroll = (evt) => {
+          const { scrollLeft: left, scrollTop: top } = evt.target;
+          displayLayer.style.left = `-${left}px`;
+          displayLayer.style.top = `-${top}px`;
+          setScrollOffset({ left, top });
```

Comment:
@barklund I was debugging this to try to understand what's happening here with the delay and thinking that perhaps it might be related to assigning `style` here directly while using the native scroll event for the edit layer?

Some things noticed:
1. Even when commenting out `scrollableLayer.scrollTop = scroll.current.top;` and `scrollableLayer.scrollTop = scrollTop;` and not changing anything else in the code, it still continues working as before (the lagging version). It seems like `scrollTop` is assigned to the `scrollableLayer` via native scroll event anyway. (Didn't test out with `scrollLeft`).
2. Initially I tried assigning the `top` value here directly to `scrollableLayer` to test it out without using `scrollTop` but since the `scrollTop` value seems to be assigned automatically anyway, then it doesn't work like it -- it got assigned double (top + scrollTop):
![test](https://user-images.githubusercontent.com/3294597/144839727-10ccb19d-dc99-4506-8695-7480e686eaf7.gif)

Also tried nulling `scrollTop` and tracking the scrolled  `top` manually, however, since it still can't be tested fully without the `scrollTop`, there's still a lag (different kind of lag though and almost not noticeable when scrolling slower than on this gif):
![testing](https://user-images.githubusercontent.com/3294597/144843946-2533e017-d139-4e18-97e3-a849f27fdd63.gif)

Tried this for scrolling down only (and commented out assigning `scrollTop` below from both `useEffect`s):
```
const onScroll = (evt) => {
          evt.preventDefault();
          evt.stopPropagation();
          const { scrollLeft: left, scrollTop: top } = evt.target;
          if (!lastScrollTopRef.current) {
            lastScrollTopRef.current = top;
          } else {
            lastScrollTopRef.current = lastScrollTopRef.current + top;
          }
          displayLayer.style.left = `-${left}px`;
          displayLayer.style.top = `-${lastScrollTopRef.current}px`;
          if (scrollableLayer && !isControlled && isScrollable) {
            scrollableLayer.scrollTop = 0;
            scrollableLayer.style.top = `-${lastScrollTopRef.current}px`;
          }
          setScrollOffset({ left, top });
          return false;
        };
```

Wondering if there's a way to unify the layers more to apply either style or `scrollTop` to both instead of using native `scrollTop` on the edit layer and `style` on the display?